perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -44,13 +47,52 @@ enum Flags
     Flag_count
 };
 
-static void SetFlags(u16 Result, bool* FlagArray)
-{
-    FlagArray[Flag_SF] = Result & 0x8000;
-    FlagArray[Flag_ZF] = (Result == 0);
-}
+static void SetFlags(const instruction& Instruction, const u16 LeftOperandValue, 
+    const u16 RightOperandValue, const u16 Result, bool* FlagArray)
+{   
+    u16 HighOrderBit = (Instruction.Operands[1].Register.Count == 1) ? 0x80 : 0x8000;
+    bool Parity = true;
+       
+    switch (Instruction.Op)
+    {
+        case Op_add: [[fallthrough]];
+        case Op_sub: 
+        {
+            if (Instruction.Op == Op_add)
+            {
+                FlagArray[Flag_CF] = ((LeftOperandValue & HighOrderBit) || (RightOperandValue & HighOrderBit))
+                    && !(Result & HighOrderBit);
+                FlagArray[Flag_AF] = ((LeftOperandValue & 0x8) || (RightOperandValue & 0x8))
+                    && !(Result & 0x8);
+                FlagArray[Flag_OF] = ((LeftOperandValue & HighOrderBit) == (RightOperandValue & HighOrderBit))
+                    && ((Result & HighOrderBit) != (LeftOperandValue & HighOrderBit));
+            }
+            else
+            {
+                FlagArray[Flag_CF] = RightOperandValue > LeftOperandValue;
+                FlagArray[Flag_AF] = (RightOperandValue & 0xF) > (LeftOperandValue & 0xF);
+                FlagArray[Flag_OF] = ((LeftOperandValue & HighOrderBit) != (RightOperandValue & HighOrderBit)) && !(Result & HighOrderBit);
+            }
+            
+        }
+        [[fallthrough]];
+        default: 
+        {
+            FlagArray[Flag_SF] = Result & HighOrderBit;
+            FlagArray[Flag_ZF] = (Result == 0);
```

Comment:
Also here I noticed that you used `size_t` and I don't see a good reason to do it. So I think my comment last time wasn't very good at explaining when it's useful.

Basically, `size_t` is a type that is guaranteed to be large enough to contain any necessary index/offset to reach for stuff in your data structures (like arrays). Here you're not doing this at all, there's no array. So there's really no particular reason to use `size_t`.

Reply:
Your last comment was perfectly clear!
As noted [here](https://github.com/darkgiggs/computer_enhance/pull/4#discussion_r1160111022) I used to iterate over a loop here and didn't change it back

Reply:
OK! :D But there's still `size_t` here

Reply:
I blame Github for not showing the actual line (and myself for not checking thoroughly enough)
c726f47da6041ed3493c5d2e5bd072a7d9523ffb